### ISSUE TITLE
Allow CA1727 to be used on custom types

### ISSUE THREAD

**Issue by EricEzaM** [dotnet/roslyn-analyzers#7605]
### Analyzer

**Diagnostic ID**: [CA1727](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1727)

### Describe the improvement

As per the code in [LoggerMessageDefineAnalyzer.cs](https://github.com/dotnet/roslyn-analyzers/blob/main/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/LoggerMessageDefineAnalyzer.cs), the analyzer works on:

```
WellKnownTypeNames.MicrosoftExtensionsLoggingLoggerExtensions
WellKnownTypeNames.MicrosoftExtensionsLoggingILogger
WellKnownTypeNames.MicrosoftExtensionsLoggingLoggerMessage
```

If you write your own extensions for logger, you cannot make use of this analyzer.

### Describe suggestions on how to achieve the rule

Allow Msbuild properties to be used to customise the behaviour. I believe this cookbook section could help: https://github.com/dotnet/roslyn/blob/main/docs/features/source-generators.cookbook.md#consume-msbuild-properties-and-metadata

Essentially allowing specification of which classes this analyzer should apply to. Not sure if this would need to use the "AdditionalFiles" item or if it could define a custom one, but essentially:

```xml
<ItemGroup>
    <AdditionalLoggerAnalyzerType>MyNamespace.MyLoggerExtensionMethods</AdditionalLoggerAnalyzerType>
</ItemGroup>
```

Then use `CompilerVisibleProperty` or ` CompilerVisibleItemMetadata` to get this information in the analyzer and allow methods from these types to be included in the analysis.
